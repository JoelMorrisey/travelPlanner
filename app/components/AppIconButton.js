import React from 'react';

import { MaterialCommunityIcons as Icon } from '@expo/vector-icons';

import { View, StyleSheet, TouchableOpacity } from 'react-native'

function AppIconButton({name, size=20, style, onPress}) {
    return (
        <TouchableOpacity onPress={onPress} activeOpacity={0.5} style={[{width: size}, style]}>
            <View style={[styles.container, {width: size, borderRadius: size/2}]}>
                <Icon name={name} size={size}/>
            </View>
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: "white",
    }
})

export default AppIconButton;