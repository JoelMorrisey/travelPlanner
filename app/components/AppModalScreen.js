import React from 'react';

import { Modal } from 'react-native'



function AppModalScreen({ children, activeControl, active, animationType='slide', backButton}) {
    //Close the module
    const closeModal = () => {
        activeControl(false);
    }    

    //Set the back button functionality
    if (!backButton) {
        backButton = () => {
            return closeModal();
        }
    }

    return (
        <Modal 
            animationType={animationType}
            visible={active}
            onRequestClose={() => {
                backButton();
            }}
            statusBarTranslucent={true}
        >
            {
                children({backButton, closeModal})
            }
        </Modal>
    );
}

export default AppModalScreen;