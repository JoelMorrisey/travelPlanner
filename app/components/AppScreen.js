import React from 'react';

import { View, SafeAreaView, StyleSheet, ImageBackground } from 'react-native';

import Constants from 'expo-constants';

function AppScreen({children, style, backgroundImage, ...other}) {
    return (
        <SafeAreaView style={[styles.screen]}>
            <OptionalBackGroundImage backgroundImage={backgroundImage}>
                <View style={[styles.view, style]} {...other}>
                    {children}
                </View>
            </OptionalBackGroundImage>
        </SafeAreaView>
    );
}

function OptionalBackGroundImage({children, backgroundImage}) {
    return (
        <>
            {
                backgroundImage ?
                    <ImageBackground source={backgroundImage} style={styles.backgroundImage}>{children}</ImageBackground>
                    :
                    children
            }
        </>
    )
}

const styles = StyleSheet.create({
    screen:{
        flex:1,
        marginTop: Constants.statusBarHeight,
    },
    view: {
        flex:1
    },
    backgroundImage: {
        flex: 1,
        width: "100%",
        height: "100%"
    }
})

export default AppScreen;