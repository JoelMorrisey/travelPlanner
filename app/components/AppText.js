//React native
import React from 'react';
import { Text, StyleSheet } from 'react-native'

function AppText({children, style, ...other}) {
    return (
        <Text style={[styles.text, style]} {...other}>
            {children}
        </Text>
    );
}

const styles = StyleSheet.create({
    text: {
        fontSize: 16
    }
})

export default AppText;