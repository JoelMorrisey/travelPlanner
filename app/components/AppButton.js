import React from 'react';

import { TouchableOpacity, StyleSheet } from 'react-native';

import AppColor from "config/AppColor";
import AppText from "components/AppText";

function AppButton({title, onPress, style, textStyle, disabled=false}) {
    return (
        <TouchableOpacity disabled={disabled} onPress={onPress} style={[disabled ? styles.disabledButton:styles.button, style]}>
            <AppText style={[styles.text, textStyle]}>{title}</AppText>
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
    button:{
        backgroundColor: AppColor.primary,
        borderRadius: 20,
        borderColor: AppColor.border,
        padding: 15,
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 30,
        width: 200,
        height: 65,
        borderWidth: 1,
    },
    disabledButton:{
        backgroundColor: "grey",
        borderRadius: 20,
        borderColor: AppColor.border,
        padding: 15,
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 30,
        width: 200,
        height: 65,
        borderWidth: 1,
    },
    text:{
        fontSize: 16,
        textTransform: 'uppercase',
        fontWeight: 'bold',
    },
})

export default AppButton;
