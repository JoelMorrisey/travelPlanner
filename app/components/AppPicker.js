//React native
import React, { useState } from 'react';
import { Button, Modal, StyleSheet, FlatList, TouchableOpacity, View } from 'react-native'

//Third party
import { MaterialCommunityIcons as Icon } from '@expo/vector-icons';

//Components
import AppText from 'components/AppText'

//Config
import AppStyles from 'config/AppStyles'

function AppPicker({title, selectionOptions, onSelect, buttonStyle, header}) {
    let [modalVisible, setModalVisible] = useState(false)
    return (
        <>
            <Modal
                animationType="slide"
                visible={modalVisible}
                style={{backgroundColor:"black"}}
                onRequestClose={() => {
                    setModalVisible(false);
                }}
            >
                <AppText style={[AppStyles.title, {alignSelf: "center", marginTop: 20}]}>{header}</AppText>
                <FlatList
                    data={selectionOptions}
                    renderItem={({ item }) => <Options title={item.name} onPress={() => {setModalVisible(false); onSelect(item)}}/>}
                    keyExtractor={item => item.id.toString()}
                />
                <Button title={"close"} onPress={() => setModalVisible(false)}/>
            </Modal>
            <TouchableOpacity onPress={() => setModalVisible(true)} style={[styles.button, buttonStyle]}>
                <AppText style={styles.text}>{title}</AppText>
                <Icon
                    name="chevron-down"
                    size={20}
                    style={{paddingRight:5}}
                />
            </TouchableOpacity>
            {/* chevron-down */}
        </>         
    );
}

//Options to select for the picker format
function Options({onPress, title}) {
    return (
        <View style={{width: "100%", alignItems: "center"}}>
            <TouchableOpacity onPress={onPress} style={{borderWidth: 2, backgroundColor: "#EEEEEE", borderColor: "white", borderBottomColor: "black", width: "85%", alignItems: "center", justifyContent: "center", marginBottom: 10, height: 50}}>
                <AppText style={[styles.text]}>{title}</AppText>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    button:{
        borderRadius: 0,
        width: "100%",
        height: 35,
        justifyContent: "space-between",
        alignItems: 'center',
        marginBottom: 30,
        borderWidth: 1,
        flexDirection: "row"
    },
    text:{
        fontSize: 16,
        textTransform: 'uppercase',
        fontWeight: 'bold',
        paddingLeft: 15
    }
})

export default AppPicker;