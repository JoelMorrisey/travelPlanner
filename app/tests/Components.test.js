import React from 'react';
import renderer from 'react-test-renderer';

//Components
import AppButton from 'components/AppButton';
import AppIconButton from 'components/AppIconButton';
import AppModalScreen from 'components/AppModalScreen';
import AppPicker from 'components/AppPicker';
import AppScreen from 'components/AppScreen';
import AppText from 'components/AppText';


//Component tests
describe('AppScreen', () => {
    let testComps = {};

    //Setup
    beforeAll(() => {
        testComps["AppScreen"] = renderer.create(<AppScreen/>).toJSON();
        testComps["AppScreenWithBackGround"] = renderer.create(<AppScreen backgroundImage={require("./testImages/testBackground.jpg")}/>).toJSON();
        testComps["AppScreenContainingAChild"] = renderer.create(<AppScreen>Hello</AppScreen>).toJSON();
    })

    //Tests
    //test no background
    it('no background - renders correctly', () => {
        expect(testComps["AppScreen"]).toMatchSnapshot();
    });

    it('no background -  does not contain image component', () => {
        expect(testComps["AppScreen"].children[0].type).toBe("View");
        expect(testComps["AppScreen"].children[0].children).toBeNull();
    });

    //test back ground
    it('with background - renders correctly', () => {
        expect(testComps["AppScreenWithBackGround"]).toMatchSnapshot();
    });
    it('with background -  contains an image component', () => {
        expect(testComps["AppScreenWithBackGround"].children[0].children[0].type).toBe("Image");
    });

    //test containing a child
    it('with child - renders correctly', () => {
        expect(testComps["AppScreenContainingAChild"]).toMatchSnapshot();
    });
    it('with child - child exists', () => {
        expect(testComps["AppScreenContainingAChild"].children[0].children.includes("Hello")).toMatchSnapshot();
    });
});

describe('AppText', () => {
    let testComps = {};

    //Setup
    beforeAll(() => {
        testComps["AppText"] = renderer.create(<AppText>Hello</AppText>).toJSON();
    })

    //Tests
    it('AppText - renders correctly', () => {
        expect(testComps["AppText"]).toMatchSnapshot();
    });

    it('has 1 child', () => {
        expect(testComps["AppText"].children.length).toBe(1);
    });

    it('has child string \"Hello\"', () => {
        expect(testComps["AppText"].children.includes("Hello")).toBeTruthy();
    }); 
});

describe('AppPicker', () => {
    let testComps = {};

    //Setup
    beforeAll(() => {
        testComps["AppPicker"] = renderer.create(
            <AppPicker
                title={"NONE"}
                selectionOptions={[
                    {
                        id: 1,
                        name: "Sleep",
                        icon: "bed",
                        size: 20
                    },
                    {
                        id: 2,
                        name: "Eat",
                        icon: "food",
                        size: 20
                    },
                    {
                        id: 3,
                        name: "Adventure",
                        icon: "biathlon",
                        size: 20
                    }
                ]}
                onSelect={(tag) => tag}
                buttonStyle={{ width: "85%" }}
                header="Pick a category"
            />
        ).toJSON();
    })

    //Tests
    //Test render
    it('AppPicker - renders correctly', () => {
        expect(testComps["AppPicker"]).toMatchSnapshot();
    });

    //assists to see where error is if error occurs with snapshot if error is to do with expected children
    it('has 2 child', () => {
        expect(testComps["AppPicker"].length).toBe(2);
        expect(testComps["AppPicker"][0].type).toBe("Modal");
        expect(testComps["AppPicker"][1].type).toBe("View");
    });
});

describe('AppModalScreen', () => {
    let testComps = {};

    //Setup
    beforeAll(() => {
        testComps["AppModalScreen - functions no specific"] = renderer.create(
            <AppModalScreen
                active={true}
                activeControl={(value) => value}
            >
                {
                    ({ backButton, closeModal }) => {
                        return `${closeModal} ${backButton}`
                    }
                }
            </AppModalScreen>
        ).toJSON();
        testComps["AppModalScreen - functions specific"] = renderer.create(
            <AppModalScreen
                active={true}
                activeControl={(value) => value}
                backButton={() => "test"}
            >
                {
                    ({ backButton, closeModal }) => {
                        return `${closeModal} ${backButton}`
                    }
                }
            </AppModalScreen>
        ).toJSON();
        testComps["AppModalScreen - active"] = renderer.create(
            <AppModalScreen
                active={true}
                activeControl={(value) => "value to set" + value}
            >
                {
                    ({ backButton }) => {
                        return "test"
                    }
                }
            </AppModalScreen>
        ).toJSON();
        testComps["AppModalScreen - not active"] = renderer.create(
            <AppModalScreen
                active={false}
                activeControl={(value) => "value to set" + value}
            >
                {
                    ({ backButton }) => {
                        return "test"
                    }
                }
            </AppModalScreen>
        ).toJSON();
    })

    //Test active AppModalScreen
    it('AppModalScreen active - renders correctly', () => {
        expect(testComps["AppModalScreen - active"]).toMatchSnapshot();
    });

    it('AppModalScreen active - has 1 child', () => {
        expect(testComps["AppModalScreen - active"].children.length).toBe(1);
        expect(testComps["AppModalScreen - active"].children[0]).toBe("test");
    });

    //Test inactive AppModalScreen
    it('AppModalScreen not active - renders correctly', () => {
        expect(testComps["AppModalScreen - not active"]).toMatchSnapshot();
    });

    it('AppModalScreen not active - has 1 child', () => {
        expect(testComps["AppModalScreen - not active"].children.length).toBe(1);
        expect(testComps["AppModalScreen - not active"].children[0]).toBe("test");
    });

    //Test when backButton function is not specified
    it('AppModalScreen functions no specific - closeModal and backbutton are the same function', () => {
        expect(testComps["AppModalScreen - functions no specific"].children.length).toBe(1);
        expect(testComps["AppModalScreen - functions no specific"]).toMatchSnapshot();
    });

    //Test when backButton function is specified
    it('AppModalScreen functions specific - closeModal and backbutton are different functions', () => {
        expect(testComps["AppModalScreen - functions specific"].children.length).toBe(1);
        expect(testComps["AppModalScreen - functions specific"]).toMatchSnapshot();
    });
});

describe('AppIconButton', () => {
    let testComps = {};

    //Setup
    beforeAll(() => {
        testComps["AppIconButton"] = renderer.create(<AppIconButton name="arrow-left" size={20} onPress={() => "test"}/>).toJSON();
    })

    //Tests
    it('renders correctly', () => {
        expect(testComps["AppIconButton"]).toMatchSnapshot();
    });
});

describe('AppButton', () => {
    let testComps = {};

    //Setup
    beforeAll(() => {
        testComps["AppIconButton - disabled"] = renderer.create(<AppButton title="test" onPress={() => "test"} disabled={true}/>).toJSON();
        testComps["AppIconButton - enabled"] = renderer.create(<AppButton title="test" onPress={() => "test"} disabled={false}/>).toJSON();
    })

    //Tests
    //Test button disabled (should mainly change style)
    it('disabled button - renders correctly', () => {
        expect(testComps["AppIconButton - disabled"]).toMatchSnapshot();
    });

    //Test button enabled (should mainly change style)
    it('enabled button - renders correctly', () => {
        expect(testComps["AppIconButton - enabled"]).toMatchSnapshot();
    });
});