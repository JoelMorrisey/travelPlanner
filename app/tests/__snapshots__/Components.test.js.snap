// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton disabled button - renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "grey",
      "borderColor": "#05750C",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 65,
      "justifyContent": "center",
      "marginBottom": 30,
      "opacity": 1,
      "padding": 15,
      "width": 200,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          undefined,
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`AppButton enabled button - renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "#05750C",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 65,
      "justifyContent": "center",
      "marginBottom": 30,
      "opacity": 1,
      "padding": 15,
      "width": 200,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          undefined,
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`AppIconButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
        },
        Object {
          "borderRadius": 10,
          "width": 20,
        },
      ]
    }
  />
</View>
`;

exports[`AppModalScreen AppModalScreen active - renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  visible={true}
>
  test
</Modal>
`;

exports[`AppModalScreen AppModalScreen functions no specific - closeModal and backbutton are the same function 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  visible={true}
>
  function closeModal() {
    activeControl(false);
  } function backButton() {
      return closeModal();
    }
</Modal>
`;

exports[`AppModalScreen AppModalScreen functions specific - closeModal and backbutton are different functions 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  visible={true}
>
  function closeModal() {
    activeControl(false);
  } function backButton() {
        return "test";
      }
</Modal>
`;

exports[`AppModalScreen AppModalScreen not active - renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  visible={false}
>
  test
</Modal>
`;

exports[`AppPicker AppPicker - renders correctly 1`] = `
Array [
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "backgroundColor": "black",
      }
    }
    visible={false}
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Array [
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginBottom": 20,
            },
            Object {
              "alignSelf": "center",
              "marginTop": 20,
            },
          ],
        ]
      }
    >
      Pick a category
    </Text>
    <RCTScrollView
      data={
        Array [
          Object {
            "icon": "bed",
            "id": 1,
            "name": "Sleep",
            "size": 20,
          },
          Object {
            "icon": "food",
            "id": 2,
            "name": "Eat",
            "size": 20,
          },
          Object {
            "icon": "biathlon",
            "id": 3,
            "name": "Adventure",
            "size": 20,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EEEEEE",
                  "borderBottomColor": "black",
                  "borderColor": "white",
                  "borderWidth": 2,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "opacity": 1,
                  "width": "85%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "paddingLeft": 15,
                        "textTransform": "uppercase",
                      },
                    ],
                  ]
                }
              >
                Sleep
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EEEEEE",
                  "borderBottomColor": "black",
                  "borderColor": "white",
                  "borderWidth": 2,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "opacity": 1,
                  "width": "85%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "paddingLeft": 15,
                        "textTransform": "uppercase",
                      },
                    ],
                  ]
                }
              >
                Eat
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EEEEEE",
                  "borderBottomColor": "black",
                  "borderColor": "white",
                  "borderWidth": 2,
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "opacity": 1,
                  "width": "85%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "paddingLeft": 15,
                        "textTransform": "uppercase",
                      },
                    ],
                  ]
                }
              >
                Adventure
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          close
        </Text>
      </View>
    </View>
  </Modal>,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 0,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "space-between",
        "marginBottom": 30,
        "opacity": 1,
        "width": "85%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingLeft": 15,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      NONE
    </Text>
  </View>,
]
`;

exports[`AppScreen no background - renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 44,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  />
</RCTSafeAreaView>
`;

exports[`AppScreen with background - renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 44,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`AppScreen with child - child exists 1`] = `true`;

exports[`AppScreen with child - renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 44,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    Hello
  </View>
</RCTSafeAreaView>
`;

exports[`AppText AppText - renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;
